src/util/loginUtil.ts

```diff
@@ -24,8 +24,11 @@ export class LoginUtil {
      * @returns true if the user needs to log in to access the cluster, and false otherwise
      */
     public async requireLogin(): Promise<boolean> {
-        return await Promise.any([Oc.Instance.getCurrentUser(), Odo.Instance.getProjects()])
-            .then(() => false)
-            .catch(() => true);
+        if (await Oc.Instance.isLoggedIn()) {
+            return await Promise.any([Oc.Instance.getCurrentUser(), Odo.Instance.getProjects()])
+                .then(() => false)
+                .catch(() => true);
```

Comment:
`Oc.Instance.isLoggedIn()` actually does the same as `Oc.Instance.getCurrentUser()`  but in not-blocking way, so probably we don't don't need this extra method, but we have to make `Oc.Instance.getCurrentUser()`  to be non-blocking instead. 

The thing I'd like to ask here... Once `oc whoami` returns the logged in user, should we continue checking whether no error happens on `odo list project` execution?

Reply:
I think we should be good to remove this function and replace it with `Oc.Instance.isLoggedIn()` everywhere it's used.